contracts/DomainStorage.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.0;
+
+/// @title Domain controller
+/// @author Serhii Bondarenko
+/// @notice You can use this contract for storing a domain name and an address of the domain owners
+/// @dev The contract owner can change the price of domains and withdraw ether from the contract
+/// @custom:experimental This is an experimental contract.
+contract DomainStorage {
+
+    address public owner;
+    uint256 public domainCost;
+    Domain[] public domains;
+
+    /// @notice Event during the registration of a new domain
+    event DomainAdded(
+        address indexed from,
+        uint indexed when,
+        string domain,
+        address domainOwner
+    );
+
+    /// @notice Domain structure that contains a domain name and the owner address
+    struct Domain {
```

Comment:
line 24 
NatSpec: Struct should be on top of storage
also Struct a bit over logic for this task
- 1 point